### DISCUSSION TITLE
Should I/is it expected to need to tune pid settings?

### DISCUSSION THREAD

**Discussion by jaguilar** [pybricks/support#1224] [Q&A] [answered]
I have a bot that seems to more or less do the things I programmed it to do. One thing I have noticed is that my medium motor has jerky movements especially at lower speeds. I can post a video tomorrow, but is this an indication that it is in need of PID tuning in my project? If jerky motion does indicate a need for PID tuning, is it typical for the motors to need PID tuning on a bot, or is it a sign of a mistake on my part?

**Comment by laurensvalk** [comment, marked as answer]
Tuning shouldn't be needed in most cases.

What speeds are you running at? Anything below 150 deg/s (or "15%" in the LEGO app) is not really recommended for this type of motor. They have quite a bit of friction, and these speeds quickly get you into a stick-slip situation.

So if you need to drive slower, usually smaller wheels is a good idea, or you can use gearing (but this may also introduce backlash).

EDIT: What platform is this on, and which version? I see you might be using EV3. Those motors can work a bit better at lower speeds, but it hasn't had all of the updates.

**Reply by jaguilar** [reply]
I'm on ev3. And, yeah, I was trying to run the motor at about 10degrees per second. :P Way under the limit you proposed.

The motor is driving the upper stage of a toy robotic arm, and I don't want the movement to be too too rapid, because the whole construction is quite flimsy and moving the arm quickly introduces a lot of wobble. (I have only the ev3 base set so my turntable options are limited.) What I'll do is add another step down gear and increase the speed proportionally and we will see how we go.

**Reply by jaguilar** [reply]
Oh, and it's pybricks 2, the latest release of that I believe. (I'm the same guy who was trying to use 3.3 even though it's not supported, but I have seen the light.)

**Reply by laurensvalk** [reply]
Right, I think the Large EV3 motors can go down to 50 deg/s, but the medium is similarly sticky as the newer ones.

As an exercise, see if you're able to manually turn it at 10 deg/s. :robot: 

> I have seen the light.

Thanks for sticking with Pybricks nonetheless :smile:

**Reply by jaguilar** [reply]
I'm not tryna do any visual programming out here. :) Python or bust.

**Reply by jaguilar** [reply]
Minor follow-up question. I've found that the medium motor, at least on my build, basically won't do anything if the configured maximum torque is lower than 30%. I guess is this is because it can't overcome its startup friction at the lower torque numbers.

Do we know what those thresholds are for the large motor? Obviously I can get this by experimenting but I'm curious if these are known facts about these motors and if so if they have been compiled anywhere where I can see them.

**Reply by laurensvalk** [reply]
It varies by motor and factors like age, but the EV3 large motors will generally move even at just 10% duty cycle / torque.

**Reply by jaguilar** [reply]
Wonderful. The medium motor doesn't seem to put excessive stress on the parts during limit sensing at 40%, but the large one was. I will try running it at 10-15% and see if that gives less plastic-bending results. :)

**Reply by jaguilar** [reply]
Just reporting back that by increasing the rotation speed of the medium motor, and by reducing the power on the large motor, I was able to get my arm to calibrate with a reasonable level of success. Thanks again!
